hera_pspec/plot.py

```diff
@@ -1009,3 +1010,1272 @@ def _round_sigfig(x, up=True):
+            y = stats.norm.pdf(bins, np.mean(data), np.std(data))
+            ax.plot(bins, y, lw=2, ls='--', label='Gaussian', color='blue')
+        if fit_curve == 'Exponential':
+            y = stats.laplace.pdf(bins, np.mean(data), np.std(data)/np.sqrt(2.))
+            ax.plot(bins, y, lw=2, ls='--', label='Exponential', color='blue')
+    
+    ax.legend(fontsize=12, loc='lower center',framealpha=0)
+    ax.set_yscale('log')
+    ax.grid()
+    
+    ax = axes[1]
+    data = d.imag
+    if plot_mode == 'normal':
+        if show_robust == True:
+            n, bins, patches = ax.hist(data, bins=11,  histtype='step', lw=3, 
+                range=((np.mean(data)-4.*np.std(data)), (np.mean(data)+4.*np.std(data)))
+                ,label='real\n(mean:{:.2f}, \nmean(r):{:.2f}, \nstd:{:.2f},\nstd(r):{:.2f})'.format(np.mean(data), astats.biweight_location(data),np.std(data), np.sqrt(astats.biweight_midvariance(data)))
+                ,color = 'blue' )
+        else:
+            n, bins, patches = ax.hist(data, bins=11,  histtype='step', lw=3, 
+                range=((np.mean(data)-4.*np.std(data)), (np.mean(data)+4.*np.std(data)))
+                ,label='real\n(mean:{:.2f}, \nstd:{:.2f})'.format(np.mean(data),np.std(data))
+                ,color = 'blue' )
+        bincenters = 0.5*(bins[1:]+bins[:-1])
+        menStd     = np.sqrt(n)
+        ax.bar(bincenters, n, width=0., ecolor='red', yerr=menStd)
+    
+    if plot_mode == 'density':
+        if show_robust == True:
+            n, bins, patches = ax.hist(data, bins=11,  histtype='step', lw=3, 
+                range=((np.mean(data)-4.*np.std(data)), (np.mean(data)+4.*np.std(data)))
+                ,label='real\n(mean:{:.2f}, \nmean(r):{:.2f}, \nstd:{:.2f},\nstd(r):{:.2f})'.format(np.mean(data), astats.biweight_location(data),np.std(data), np.sqrt(astats.biweight_midvariance(data)))
+                ,color = 'blue', density=True)
+        else:
+            n, bins, patches = ax.hist(data, bins=11,  histtype='step', lw=3, 
+                range=((np.mean(data)-4.*np.std(data)), (np.mean(data)+4.*np.std(data)))
+                ,label='real\n(mean:{:.2f}, \nstd:{:.2f})'.format(np.mean(data),np.std(data))
+                ,color = 'blue', density=True)
+        
+        assert fit_curve in ['Gaussian', 'Exponential'],\
+        "'%s' not a valid axis; use 'Gaussian' or 'Exponential'" % fit_curve
+        if fit_curve == 'Gaussian':
+            y = stats.norm.pdf(bins, np.mean(data), np.std(data))
+            ax.plot(bins, y, lw=2, ls='--', label='Gaussian', color='blue')
+        if fit_curve == 'Exponential':
+            y = stats.laplace.pdf(bins, np.mean(data), np.std(data)/np.sqrt(2.))
+            ax.plot(bins, y, lw=2, ls='--', label='Exponential', color='blue')
+    
+    ax.legend(fontsize=12, loc='lower center',framealpha=0)
+    ax.set_yscale('log')
+    ax.grid()
+
+
+def plot_error(uvp, uvp_td, key, time_index, Tsys, normalized=True, extra_error_types=[], **kwargs):
```

Comment:
This is slightly redundant with `plot.delay_spectrum`, which can plot errorbars by using the `plot_stats` kwarg. Can you see if that function satisfies what you want to do here, just to minimize redundant code? If not we can keep this function and get rid of `delay_spectrum(plot_stats)` kwarg.

Reply:
plot_error() tries to plot different kinds of error bars without delay spectra at the same time, and asks for a outer time-difference power spectrum and cov_arrays. While plot_stats asks for the data stored in stats_array. Might be better to keep both?